libsupport/include/katana/Result.h

```diff
@@ -283,11 +350,13 @@ extract_result_value(Result<void>&&) {
 #define KATANA_CHECKED_IMPL(result_name, expression, ...)                      \
   ({                                                                           \
     auto result_name = (expression);                                           \
-    if (!result_name) {                                                        \
-      return result_name.error().WithContext(__VA_ARGS__);                     \
+    if (::katana::internal::CheckedExpressionFailed(result_name)) {            \
+      return ::katana::internal::CheckedExpressionToError(result_name)         \
+          .WithContext(__VA_ARGS__)                                            \
+          .WithContext("({}:{})", __FILE__, __LINE__);                         \
```

Comment:
This is crying out for a better way to handle it, while still keeping a stack trace